prefect_github/repository.py

```diff
@@ -23,6 +31,121 @@
 return_fields_defaults = initialize_return_fields_defaults(config_path)
 
 
+class GitHubRepository(ReadableDeploymentStorage):
+    """
+    Interact with files stored on GitHub repositories.
+    """
+
+    _block_type_name = "GitHub Repository"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/187oCWsD18m5yooahq1vU0/ace41e99ab6dc40c53e5584365a33821/github.png?h=250"  # noqa: E501
+
+    repository: str = Field(
+        default=...,
+        description=(
+            "The URL of a GitHub repository to read from, in either HTTPS or SSH "
+            "format. If you are using a private repo, it must be in the HTTPS format."
+        ),
+    )
```

Comment:
Can we change this attribute to `repository_url` so that it isn't confused with repository name?

```suggestion
    repository_url: str = Field(
        default=...,
        title="Repository URL",
        description=(
            "The URL of a GitHub repository to read from, in either HTTPS or SSH "
            "format. If you are using a private repo, it must be in the HTTPS format."
        ),
    )
```